### PR TITLE
Filtering SupportedTargetFrameworkAlias based on target framework moniker

### PR DESCRIPTION
**Pull request by sfoslund**
- Adding filtering based on matching TargetFrameworkIdentifiers
- Moving the GenerateSupportedTargetFrameworkAlias target to be later in evaluation (See also https://github.com/dotnet/msbuild/pull/5753)

### CODE REVIEW

File: src/Tasks/Microsoft.NET.Build.Tasks.UnitTests/GivenThatWeWantToGenerateSupportedTargetFrameworkAlias.cs [hunk 1]
```diff
@@ -25,24 +25,40 @@ private static List<(string targetFrameworkMoniker, string displayName)> MockSup
                 ( ".NETFramework,Version=v4.8", ".NET Framework 4.8"),
             };
```
Comment (davkean) [inline src/Tasks/Microsoft.NET.Build.Tasks.UnitTests/GivenThatWeWantToGenerateSupportedTargetFrameworkAlias.cs:27]: Can you add a test case where the TFM is lower case? I want to see if NuGet normalizes it, if not, you need an OrdinalIgnoreCase check as TFMs are not case sensitive.

Comment (sfoslund) [reply src/Tasks/Microsoft.NET.Build.Tasks.UnitTests/GivenThatWeWantToGenerateSupportedTargetFrameworkAlias.cs:27]: Good catch, thank you. I've added a test case.

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.DefaultItems.targets [hunk 2]
```diff
@@ -375,4 +375,20 @@ Copyright (c) .NET Foundation. All rights reserved.
         Condition="'$(TargetFrameworkIdentifier)' == '.NETCoreApp' and $([MSBuild]::VersionGreaterThanOrEquals($(_TargetFrameworkVersionWithoutV), '5.0')) and '$(_MicrosoftWindowsDesktopSdkImported)' == 'true' and '$(TargetFrameworks)' == ''">
     <NETSdkWarning ResourceName="UnnecessaryWindowsDesktopSDK" />
   </Target>
+  
+  <UsingTask TaskName="GenerateSupportedTargetFrameworkAlias" AssemblyFile="$(MicrosoftNETBuildTasksAssembly)" />
+
+  <!--
+    Generate a list of valid TargetFramework aliases for the project system to consume for the target framework properties page drop down
+  -->
+  <Target Name="GenerateSupportedTargetFrameworkAlias"
```
Comment (dsplaisted) [inline src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.DefaultItems.targets:384]: I'd try to find a better place to put this than in DefaultItems.targets.  That target file is for dealing with the default globs.

**Review by dsplaisted — APPROVED** [review]

**Review by melytc — APPROVED** [review]
LGTM 👍

**Pull request merged**